core/src/main/java/com/datastax/oss/driver/api/core/config/DriverExecutionProfile.java

```diff
@@ -201,4 +219,109 @@ default Duration getDuration(@NonNull DriverOption option, @Nullable Duration de
    */
   @NonNull
   SortedSet<Map.Entry<String, Object>> entrySet();
+
+  @NonNull
+  @Override
+  default DriverExecutionProfile withBoolean(@NonNull DriverOption option, boolean value) {
+    return DerivedExecutionProfile.with(this, option, value);
+  }
```

Comment:
I realized the logic for derived profiles could be extracted in a generic implementation, so now it doesn't have to be done for every implementation, which is nice.

I still kept the overrides for the Typesafe implementation, because it can be done a bit better there.